bluemira/power_cycle/net.py

```diff
@@ -0,0 +1,644 @@
+            Type of load
+        unit:
+            return unit, defaults to [W]
+        end_time:
+            for unnormalised loads this assures the load is
+            applied at the right point in time
+        """
+        timeseries, end_time = self._normalise_timeseries(timeseries, end_time)
+        load_type = LoadType.from_str(load_type)
+        return {
+            load.name: load.interpolate(timeseries, end_time, load_type)
+            if unit is None
+            else raw_uc(
+                load.interpolate(timeseries, end_time, load_type),
+                load.unit,
+                unit,
+            )
+            for load in self.loads.values()
+        }
+
+    def load_total(
+        self,
+        timeseries: np.ndarray,
+        load_type: Union[str, LoadType],
+        unit: Optional[str] = None,
+        end_time: Optional[float] = None,
+    ) -> np.ndarray:
+        """Total load for each timeseries point for a given load_type
+
+        Parameters
+        ----------
+        timeseries:
+            time array
+        load_type:
+            Type of load
+        unit:
+            return unit, defaults to [W]
+        end_time:
+            for unnormalised loads this assures the load is
+            applied at the right point in time
+        """
+        return np.sum(
+            list(
+                self.get_load_data_with_efficiencies(
+                    timeseries, load_type, unit, end_time
+                ).values()
+            ),
+            axis=0,
+        )
+
+
+@dataclass
+class Phase:
+    """Phase container"""
+
+    config: PhaseConfig
+    subphases: Dict[str, SubPhaseConfig]
+    loads: Loads
+
+    def __post_init__(self):
+        """Validate duration"""
+        if self.duration < 0:
+            raise ValueError(
+                f"{self.config.name} phase duration must be positive: {self.duration}s"
+            )
+
+    @property
+    def duration(self):
+        """Duration of phase"""
+        return getattr(np, self.config.operation)([
+            s_ph.duration for s_ph in self.subphases.values()
+        ])
+
+
+class PulseDictType(TypedDict):
+    """Pulse dictionary typing"""
+
+    repeat: int
+    data: Dict[str, Phase]
+
+
+class LibraryConfig:
+    """Power Cycle Configuration"""
+
+    def __init__(
+        self,
+        scenario: ScenarioConfig,
+        pulse: Dict[str, PulseConfig],
+        phase: Dict[str, PhaseConfig],
+        subphase: Dict[str, SubPhaseConfig],
+        system: Dict[str, SystemConfig],
+        subsystem: Dict[str, SubSystemConfig],
+        loads: Dict[str, LoadConfig],
+        durations: Optional[Dict[str, float]] = None,
+    ):
+        self.scenario = scenario
+        self.pulse = pulse
+        self.phase = phase
+        self.subphase = subphase
+        self.system = system
+        self.subsystem = subsystem
+        self.loads = loads
+        if durations is not None:
+            self._import_subphase_duration(durations)
```

Comment:
The only problem with this is if durations aren't given and there are str's in the duration params, they won't be checked and will error eventaully when called.

IMO it should be something like, check if any duration is a str value, if so get the value from the durations dict (error if not there or durations dict isn't given) and set it